internal/server/data/data.go

```diff
@@ -32,22 +32,31 @@ func NewDB(connection gorm.Dialector, loadDBKey func(db GormTxn) error) (*DB, er
 		return nil, fmt.Errorf("db conn: %w", err)
 	}
 	dataDB := &DB{DB: db}
+	tx, err := dataDB.Begin(context.TODO())
+	if err != nil {
+		return nil, err
+	}
 
 	opts := migrator.Options{
 		InitSchema: initializeSchema,
-		LoadKey: func(tx migrator.DB) error {
+		LoadKey: func(_ migrator.DB) error {
```

Comment:
Why not take this as a `tx` arg rather than using the transaction from the outside scope? I believe we control this migrator library so we could change what migrator.Options LoadKey expects.

Reply:
Good question! The problem is that gorm has already been removed from the migrator package. It now accepts only this `DB` interface, which matches the stdlib `database/sql.DB`. We can't use the `data.GormTxn` interface in the migrator package because that would cause an import cycle, and I don't really want to move the `GormTxn` into the migrator pacakge.

I think to fix this all we need is to convert the two queries that are used by `loadDBKey` to the stdlib interface. I'll do that today, and that should allow us to resolve this TODO and use the passed in transaction.

Reply:
I've got a fix for that here: https://github.com/infrahq/infra/pull/3168/commits/b2c4a68e56de236a609e19a391876fc923439692

Reply:
Thanks, I'll re-review this PR when #3168 gets in